### ISSUE TITLE
loaders: Unexpected Token When Fetching Contract Data from Etherscan

### ISSUE THREAD

**Issue by yohamta** [shazow/whatsabi#136]
I encountered an error while using `EtherscanABILoader`. The error occurs when invoking `getContract` for an unverified contract address.

#### Error message
```sh
MultiABILoaderError: MultiABILoader getContract error: EtherscanABILoader getContract error: Unexpected token 'C', "Contract s"... is not valid JSON
    at MultiABILoader.getContract (/path-to-local-env/node_modules/@shazow/whatsabi/src/loaders.ts:126:23)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at autoload (/path-to-local-env/node_modules/@shazow/whatsabi/src/auto.ts:232:40)

```

#### Reproducible Steps:
Try fetching the contract data for an unverified contract address (e.g., `0xf081470f5C6FBCCF48cC4e5B82Dd926409DcdD67`).

Example response from Etherscan API:
```json
{
   "status":"1",
   "message":"OK",
   "result":[
      {
         "SourceCode":"",
         "ABI":"Contract source code not verified",
         "ContractName":"",
         "CompilerVersion":"",
         "OptimizationUsed":"",
         "Runs":"",
         "ConstructorArguments":"",
         "EVMVersion":"Default",
         "Library":"",
         "LicenseType":"Unknown",
         "Proxy":"0",
         "Implementation":"",
         "SwarmSource":""
      }
   ]
}
```

**WhatsABI version:** `^0.15.`

**Additional note:**
Regarding the handling of ContractResult for unverified contracts, I have a small suggestion for the `autoload` function. By doing this, we can fall back to the inferred ABI in cases where an empty contract is returned.

```diff
            if (config.loadContractResult) {
                const contractResult = await loader.getContract(address);
-                if (contractResult) {
+                if (contractResult && Array.isArray(contractResult.abi) && contractResult.abi.length > 0;) {
                    // We assume that a verified contract ABI contains all of the relevant resolved proxy functions
                    // so we don't need to mess with resolving facets and can return immediately.
                    result.contractResult = contractResult;
                    result.abi = contractResult.abi;
                    result.abiLoadedFrom = contractResult.loader;
                    return result;
                }
            } else {
```

This seems to be necessary because `MultiABILoader` returns the `emptyContract` in case any ABILoader didn't return a successful result.
https://github.com/shazow/whatsabi/blob/bbb0cfdeec35bc16e2af1b0b54098c73b6b65076/src/loaders.ts#L115-L133

**Comment by shazow** [comment]
Fixed in #137, thanks for the issue and PR. :)

**Comment by shazow** [comment]
Released in v0.15.2
